classes/Attachment.php

```diff
@@ -239,25 +289,63 @@ public static function attachToProduct($idProduct, $array)
      */
     public static function getProductAttached($idLang, $list)
     {
+        if (!is_array($list)) {
+            return false;
+        }
+
         $idsAttachments = [];
-        if (is_array($list)) {
-            foreach ($list as $attachment) {
-                $idsAttachments[] = $attachment['id_attachment'];
-            }
+        foreach ($list as $attachment) {
+            $idsAttachments[] = $attachment['id_attachment'];
+        }
```

Comment:
You can also replace this with:

```suggestion
        $idsAttachments = array_column($list, 'id_attachment');
```